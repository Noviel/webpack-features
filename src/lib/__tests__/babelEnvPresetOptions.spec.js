import createBEPO from '../babelEnvPresetOptions';

describe('createBabelEnvPresetOptions', () => {
  it('should set correct default values for unspecified parameters', () => {
    const result = createBEPO({
      target: {
        name: 'browsers',
        value: true,
      },
    });
    //expect(result.useBuiltIns).toBe(true);
    expect(result.modules).toBe(false);
  });

  it('should use modern browsers list for `true` value', () => {
    const result = createBEPO({
      target: {
        name: 'browsers',
        value: true,
      },
    });
    expect(result.targets.browsers).toMatchSnapshot();
  });

  it('should set correct legacy browsers targets', () => {
    const result = createBEPO({
      target: {
        name: 'browsers',
        value: 'legacy',
      },
    });
    expect(result.targets.browsers).toMatchSnapshot();
  });

  it('should set correct modern browsers targets', () => {
    const result = createBEPO({
      target: {
        name: 'browsers',
        value: 'modern',
      },
    });
    expect(result.targets.browsers).toMatchSnapshot();
  });

  it('should set a correct node target', () => {
    const result = createBEPO({
      target: {
        name: 'node',
        value: '7.2',
      },
    });
    expect(result.targets.node).toMatchSnapshot();
  });

  it('should set a default node target for `true`', () => {
    const result = createBEPO({
      target: {
        name: 'node',
        value: true,
      },
    });
    expect(result.targets.node).toMatchSnapshot();
  });

  it('should set commonjs modules if node is used as a target', () => {
    const result = createBEPO({
      target: {
        name: 'node',
        value: 'current',
      },
    });
    expect(result.modules).toBe('commonjs');
  });

  it('should uses custom browsers list', () => {
    const customBrowsersList = 'custom browsers list';
    const result = createBEPO({
      target: {
        name: 'browsers',
        value: customBrowsersList,
      },
    });
    expect(result.targets.browsers).toBe(customBrowsersList);
  });
});
