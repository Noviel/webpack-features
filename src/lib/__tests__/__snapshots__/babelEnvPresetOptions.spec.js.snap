// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createBabelEnvPresetOptions should set a correct node target 1`] = `"7.2"`;

exports[`createBabelEnvPresetOptions should set a default node target for \`true\` 1`] = `"8.4"`;

exports[`createBabelEnvPresetOptions should set correct legacy browsers targets 1`] = `
Array [
  "> 1%",
  "last 2 versions",
  "Firefox ESR",
]
`;

exports[`createBabelEnvPresetOptions should set correct modern browsers targets 1`] = `
Array [
  ">0.25%",
  "not ie 11",
  "not op_mini all",
]
`;

exports[`createBabelEnvPresetOptions should use modern browsers list for \`true\` value 1`] = `
Array [
  ">0.25%",
  "not ie 11",
  "not op_mini all",
]
`;
